transformations/tests/test_single_column_coalesced.py

```diff
@@ -192,6 +192,81 @@ def test_single_column_coalesced_vector_notation(frontend, horizontal, vertical)
     assert len(kernel_calls) == 1
     assert kernel_calls[0].name == 'compute_column'
 
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_single_column_coalesced_masked_statement(frontend, horizontal, vertical):
+    """
+    Test resolving of vector notation in kernel and re-insertion of the
+    horizontal loop in the "driver".
+    """
+
+    fcode_driver = """
+  SUBROUTINE column_driver(nlon, nz, q, t, nb)
+    INTEGER, INTENT(IN)   :: nlon, nz, nb  ! Size of the horizontal and vertical
+    REAL, INTENT(INOUT)   :: t(nlon,nz,nb)
+    REAL, INTENT(INOUT)   :: q(nlon,nz,nb)
+    INTEGER :: b, start, end
+
+    start = 1
+    end = nlon
+    do b=1, nb
+      call compute_column(start, end, nlon, nz, q(:,:,b), t(:,:,b))
+    end do
+  END SUBROUTINE column_driver
+"""
+
+    fcode_kernel = """
+  SUBROUTINE compute_column(start, end, nlon, nz, q, t)
+    INTEGER, INTENT(IN) :: start, end  ! Iteration indices
+    INTEGER, INTENT(IN) :: nlon, nz    ! Size of the horizontal and vertical
+    REAL, INTENT(INOUT) :: t(nlon,nz)
+    REAL, INTENT(INOUT) :: q(nlon,nz)
+    INTEGER :: jk
+    REAL :: c
+
+    c = 5.345
+    DO jk = 2, nz
+      WHERE (q(start:end, jk) > 1.234)
+        q(start:end, jk) = q(start:end, jk-1) + t(start:end, jk) * c
+      ELSEWHERE
```

Comment:
Just out of curiosity: Does this have any issues with masked statements that don't have an `ELSEWHERE`?